LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -20,6 +20,7 @@ public class DefaultLoRaDataRequestHandler : ILoRaDataRequestHandler
     {
         private readonly NetworkServerConfiguration configuration;
         private readonly ILoRaDeviceFrameCounterUpdateStrategyProvider frameCounterUpdateStrategyProvider;
+        private readonly IConcentratorDeduplication concentratorDeduplication;
```

Comment:
Again: can't add comments on lines that were not changed, hence adding it here:

https://github.com/Azure/iotedge-lorawan-starterkit/pull/937/files?diff=unified&w=0#diff-68a0d3f9eca3be1aa5e03c4aac5d01e4cab390524ba9cc53e5b59bbd3746b7ccR195

Line 195: 

`var validFcntUp = isFrameCounterFromNewlyStartedDevice || (payloadFcntAdjusted > loRaDevice.FCntUp);`

How are we getting a validFcntUp ever with the same message? I'm confused how that will work, as the FCntUp on the device should reflect the last message sent. If we already processed the message, we would have that state in there?

Reply:
> How are we getting a validFcntUp ever with the same message?

Not sure how/when this can happen. Do you see something breaking that we introduced here?

Reply:
Please review 91aeb6300edbe6b0331ec51963c206bc47b0f38c

Reply:
Has changed quite a bit since the comment and now is part of the test coverage - resolving this.